commit 334278b429b17696030df742115bd1140eb4cd39
Author: Mark Emlyn David Thomas <markt@apache.org>
Date:   Sat May 1 15:44:33 2010 +0000

    Make md volatile to be sure there are no threading issues
    
    git-svn-id: https://svn.apache.org/repos/asf/tomcat/trunk@940057 13f79535-47bb-0310-9956-ffa450edef68

diff --git a/java/org/apache/catalina/realm/RealmBase.java b/java/org/apache/catalina/realm/RealmBase.java
index 44f57ee..ed8547f 100644
--- a/java/org/apache/catalina/realm/RealmBase.java
+++ b/java/org/apache/catalina/realm/RealmBase.java
@@ -112,7 +112,7 @@ public abstract class RealmBase extends LifecycleBase
     /**
      * The MessageDigest object for digesting user credentials (passwords).
      */
-    protected MessageDigest md = null;
+    protected volatile MessageDigest md = null;
 
 
     /**
